components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/cache/BaseCache.java

```diff
@@ -182,6 +182,52 @@ public void addToCache(K key, V entry, int tenantId) {
         }
     }
 
+    /**
+     * Add a cache entry if the same value is not already there.
+     *
+     * @param key   Key which cache entry is indexed.
+     * @param entry Actual object where cache entry is placed.
+     */
+    public void addToCacheIfNoDuplicate(K key, V entry, String tenantDomain) {
+
+        if (!isEnabled()) {
+            return;
+        }
+
+        try {
+            startTenantFlow(tenantDomain);
+            Cache<K, V> cache = getBaseCache();
+            if (cache != null) {
+                cache.putIfNoDuplicate(key, entry);
+            }
+        } finally {
+            PrivilegedCarbonContext.endTenantFlow();
+        }
+    }
+
+    /**
+     * Add a cache entry if the same value is not already there.
+     *
+     * @param key   Key which cache entry is indexed.
+     * @param entry Actual object where cache entry is placed.
+     */
+    public void addToCacheIfNoDuplicate(K key, V entry, int tenantId) {
+
+        if (!isEnabled()) {
+            return;
+        }
+
+        try {
+            startTenantFlow(tenantId);
+            Cache<K, V> cache = getBaseCache();
+            if (cache != null) {
+                cache.putIfNoDuplicate(key, entry);
```

Comment:
Is this a method given by baseCache or something newly implemented?

Reply:
it's newly implemented

Reply:
https://github.com/wso2/carbon-kernel/blob/089209a1b3aaa4069e15ddc503a3209f94601638/core/javax.cache/src/main/java/org/wso2/carbon/caching/impl/CacheImpl.java#L404